tfcompile_study/experiments/conv_fc_adam/test_run.py

```diff
@@ -0,0 +1,45 @@
+import tensorflow as tf
+import sys
+sys.path.insert(0, "../../.")
+from utils import run
+import pytest
+
+HIDDEN_LAYERS = 2
+LABELS = 10
+HIDDEN_SIZE = 256 if HIDDEN_LAYERS > 1 else LABELS
+LR = 0.01
+BATCH_SIZE = 64
+HEIGHT = 28
+WIDTH = 28
+CHANNELS = 3
+
+
+def model_fn(features, labels, is_training=True):
+    with tf.variable_scope("conv_fc", use_resource=True):
+        net = tf.keras.layers.Conv2D(filters=4,
+                                     kernel_size=[2, 2],
+                                     activation=tf.nn.relu)(features)
+        net = tf.layers.flatten(net)
+        for i in range(HIDDEN_LAYERS - 1):
+            net = tf.keras.layers.Dense(units=HIDDEN_SIZE,
+                                        name="hidden" + str(i),
+                                        activation=tf.nn.relu)(net)
+        logits = tf.keras.layers.Dense(units=LABELS)(net)
+        cross_entropy = tf.reduce_mean(
+            tf.nn.softmax_cross_entropy_with_logits(labels=labels,
+                                                    logits=logits))
+        train_step = tf.train.AdamOptimizer(
+            LR, name="final_node").minimize(cross_entropy)
+        with tf.control_dependencies([train_step]):
+            return tf.identity(cross_entropy, name="results")
+
+
+def input_fn():
```

Comment:
I'm not sure it is helpful to structure things in this way `input_fn` vs. just providing the inputs as part of the test via fixtures or as shape specifications

Reply:
replaced with fixtures for the inputs.

Reply:
I don’t think these are fixtures, are they?